mlx/c/ops.cpp

```diff
@@ -1037,10 +1046,6 @@ mlx_take(mlx_array a, mlx_array indices, int axis, mlx_stream s) {
   RETURN_MLX_C_ARRAY(mlx::core::take(a->ctx, indices->ctx, axis, s->ctx));
 }
 extern "C" mlx_array
-mlx_take_all(mlx_array a, mlx_array indices, mlx_stream s) {
-  RETURN_MLX_C_ARRAY(mlx::core::take(a->ctx, indices->ctx, s->ctx));
-}
```

Comment:
Should this be removed?  It looks like we still need it:

```
      "take",
      [](const array& a,
         const std::variant<int, array>& indices,
         const std::optional<int>& axis,
         StreamOrDevice s) {
        if (auto pv = std::get_if<int>(&indices); pv) {
          return axis ? take(a, *pv, axis.value(), s) : take(a, *pv, s);
        } else {
          auto indices_ = std::get<array>(indices);
          return axis ? take(a, indices_, axis.value(), s)
                      : take(a, indices_, s);
        }
      },
```

It supports th case where no axis was specified.  Potentially `mlx_take` could take an `optional_int axis` instead.

Reply:
Never mind -- per conversation in slack the implementation of no axis is trivial and the client can do it (from c++):

```
array take(const array& a, int index, StreamOrDevice s /* = {} */) {
  return take(reshape(a, {-1}, s), index, 0, s);
}
```

Reply:
This should not have been removed. I added it back by tweaking the c generator, which got confused because one `take()` overload was added on the C++ side.